src/realm/query_engine.hpp

```diff
@@ -406,6 +406,64 @@ class IntegerNodeBase : public ColumnNodeBase {
 };
 
 
+template <class LeafType>
```

Comment:
Nice, I've been wanting to do this as well. A great improvement for time series data.